File: ex05/taylor.c
#include <stdio.h>

int main() {
    double x;
    int n;

    scanf("%lf", &x);
    scanf("%d", &n);

    double soma = 0.0;
    double x_potencia = 1.0;
    double fatorial = 1.0;

    for (int i = 0; i < n; i++) {
        if (i == 0) {
            soma = x;
        } else {
            // Calculando a potência de dois em dois
            x_potencia *= (x * x);
            
            // Calculando fatorial a partir da base anterior
            int base_anterior = 2 * (i - 1) + 1;
            fatorial *= (base_anterior + 1) * (base_anterior + 2);
            
            double termo = x_potencia / fatorial;
            
            // Alternando o sinal
            if (i % 2 == 1) {
                soma -= termo;
            } else {
                soma += termo;
            }
        }
    }

    printf("%.10lf\n", soma);

    return 0;
}

File: ex05/taylor.rs
use std::io;

fn main() {
    let mut input = String::new();
    
    // Reading the first line (x)
    io::stdin().read_line(&mut input).expect("Fail to read x");
    let x: f64 = input.trim().parse().expect("x should be a floating-point number");
    
    input.clear();
    
    // Reading the second line (n)
    io::stdin().read_line(&mut input).expect("Fail to read n");
    let n: i32 = input.trim().parse().expect("n should be an integer");

    let mut soma: f64 = 0.0;
    let mut x_potencia: f64 = 1.0;
    let mut fatorial: f64 = 1.0;

    for i in 0..n {
        if i == 0 {
            soma = x;
        } else {
            x_potencia *= x * x;
            
            let base_anterior = (2 * (i - 1) + 1) as f64;
            fatorial *= (base_anterior + 1.0) * (base_anterior + 2.0);
            
            let termo = x_potencia / fatorial;
            
            // Alternating the sign
            if i % 2 == 1 {
                soma -= termo;
            } else {
                soma += termo;
            }
        }
    }

    println!("{:.10}", soma);
}
